lib/src/data/course.dart

```diff
@@ -0,0 +1,29 @@
+import "utils.dart";
+import "chat.dart";
+
+extension type CourseId(String value) {}
+
+class Course {
+  final String department;
+  final int course_number;
+  final String subject;
```

Comment:
What would this be?

Reply:
Shouldn't we specify the subject of the course in the data class? For example, CS471 would have subject "Computer Science".